### ISSUE TITLE
Training bug of mAP of Faster RCNN ,  is there anyone meet it？

### ISSUE THREAD

**Issue by yunfanLu** [dmlc/gluon-cv#258]

**Comment by yunfanLu** [comment]
https://paste.ubuntu.com/p/yT62YbGXFX/

this is the training log. 

mAP=0.0005084129989742343  when  epoch in 14, while the 13th epoch is 
 mAP=0.7135185843486027 

the max  mAP is 0.727 (0.7273405346456079) but not 78.3 in  gluon-cv.mxnet.io

**Comment by zhreshold** [comment]
We have only verified that the current hyper-parameters with batch size 4 is good, with bs 2, you might need to tune the hyper parameters a little bit.

**Comment by yunfanLu** [comment]
https://discuss.gluon.ai/t/topic/7779

Thanks for your attention，I will tune the hyper parameters a little bit and  complete the table

**Comment by ijkguo** [comment]
I noticed that in epoch 14 loss exploded. I run the experiment again on my end using the latest master gluon-cv but I did not see anything abnormal. For epoch 14 I get mAP 77.96. Epoch 14 actually decreases learning rate so there should be no explosion at all.

**Comment by FishYuLi** [comment]
I have met exactly the same problem!!! I've struggled with this problem for several days.
It is normal at the beginning, but in some epoch, the loss goes up suddenly, and the validation MAP turns to 0.  I tried to change the seed, take out the seed, or turn off the hybridize, but this problem will always occur. The only difference is that it occurs early in epoch 1 or 2, or it occurs later in maybe epoch 14 or 16.

And another strange thing, I run exactly the same code on COCO, and it works fine. Nothing goes wrong.
It only happened on VOC.

@yunfanLu Have you solved the problem? 

I use mxnet 1.3.1. Python 2.7. cuda 8.0, cudnn 5.1.3.
The log is as follows:
It works fine in epoch 13.
![Fine in epoch 13](https://ws3.sinaimg.cn/large/006tNc79ly1fz0c6qw3o9j31hd0u012p.jpg)

And the loss goes up in epoch 14.
![Wrong 14](https://ws3.sinaimg.cn/large/006tNc79ly1fz0c81805dj31hd0u0tn2.jpg)

The validation MAP turns to 0 after the loss goes wrong.
![map 0](https://ws1.sinaimg.cn/large/006tNc79ly1fz0c8zqqa7j31hd0u0qcb.jpg)

@zhreshold @ijkguo @pluskid @zheng-da Any suggestions how should I solve this problem? Any tips?

**Comment by zhreshold** [comment]
@yunfanLu In the previous session we have tried multiple times to reproduce the problem, but without luck. Guess it may relate to one of ECC memory/hardware/CUDA/CUDNN/floating point, but without reproduction it's hard to determine the root cause.

We are trying to add some checking mechanisms, in the meanwhile you can insert mAP check after each epoch and retry if it goes wrong.

**Comment by FishYuLi** [comment]
@zhreshold Ok, thank you very much. I've run the model for 7 or 8 times, mAP is checked after each epoch, and I can reproduce it every time. So is there anything I can do now?

**Comment by FishYuLi** [comment]
After updating mxnet to the nightly built version 1.5.1, this problem disappears.
